cobald_tests/daemon/core/test_config.py

```diff
@@ -95,10 +151,141 @@ def test_load_mixed_creation(self):
+                    """
+                )
+            with load(config.name) as config:
+                tagged = get_config_section(config, "__config_test__")["tagged"]
+                assert isinstance(tagged, TagTracker)
+                assert tagged.kwargs["host"] == "127.0.0.1"
+                assert tagged.kwargs["port"] == 1234
+                assert tagged.kwargs["algorithm"] == "HS256"
+                assert tagged.kwargs["users"][0]["user_name"] == "tardis"
+                assert tagged.kwargs["users"][0]["scopes"] == ["user:read"]
+
+    def test_load_tags_nested(self):
+        """Load !Tags with nested !Tags"""
+        with NamedTemporaryFile(suffix=".yaml") as config:
+            with open(config.name, "w") as write_stream:
+                write_stream.write(
+                    """
+                    pipeline:
+                        - !MockPool
+                    __config_test__:
+                        tagged: !TagTracker
+                          host: 127.0.0.1
+                          port: 1234
+                          algorithm: HS256
+                          users: !TagTracker
+                            - user_name: tardis
+                              scopes:
+                                - user:read
+                    """
+                )
+            with TagTracker.scope():
+                with load(config.name) as config:
+                    top_tag = get_config_section(config, "__config_test__")["tagged"]
+                    assert top_tag.kwargs["host"] == "127.0.0.1"
+                    assert top_tag.kwargs["port"] == 1234
+                    assert top_tag.kwargs["algorithm"] == "HS256"
+                    sub_tag = top_tag.kwargs["users"]
+                    assert isinstance(sub_tag, TagTracker)
+                    assert sub_tag.args[0]["scopes"] == ["user:read"]
+
+    def test_load_tags_eager(self):
+        """Load !Tags with substructure, immediately using them"""
+        with NamedTemporaryFile(suffix=".yaml") as config:
+            with open(config.name, "w") as write_stream:
+                write_stream.write(
+                    """
+                    pipeline:
+                        - !MockPool
+                    __config_test__:
+                        tagged: !TagTracker
+                          top: "top level value"
+                          nested:
+                            - leaf: "leaf level value"
+                    """
+                )
+            with load(config.name) as config:
+                tagged = get_config_section(config, "__config_test__")["tagged"]
+                assert isinstance(tagged, TagTracker)
+                assert tagged.orig_kwargs["top"] == "top level value"
+                assert isinstance(tagged.orig_kwargs["nested"], list)
+                assert len(tagged.orig_kwargs["nested"]) > 0
+                assert tagged.orig_kwargs["nested"] == [{"leaf": "leaf level value"}]
```

Comment:
Do you have to use `orig_kwargs` here or would `kwargs` also work?

Reply:
I need ``orig_kwargs`` to verify that all the data was already there during YAML parsing.